### PR TITLE
fix(kb): allow unicode collection names

### PR DESCRIPTION
**Pull request by bluefish-08**
## Summary
- allow Unicode characters in knowledge base collection names
- keep the existing path-safety checks while rejecting invalid or unsafe names
- add regression coverage for Unicode collection names in config and KB API tests

### CODE REVIEW

File: src/xagent/web/config.py [hunk 1]
```diff
@@ -86,9 +86,7 @@
 # Maximum file size (100MB)
 MAX_FILE_SIZE = 100 * 1024 * 1024
 
-# Allowed characters for collection and folder names (alphanumeric, underscore, hyphen)
```
Comment (rogercloud) [inline src/xagent/web/config.py]: ### [Minor] Removed comment without replacement

The previous code had a helpful comment explaining what characters were allowed and why:

```python
# Allowed characters for collection and folder names (alphanumeric, underscore, hyphen)
# This prevents path traversal and other security issues
```

This was removed entirely. `\w` is less self-documenting than an explicit character class. Consider adding a brief comment, e.g.:

```python
# Word characters (\w = letters/digits/underscore in any language) and hyphens
ALLOWED_NAME_PATTERN = re.compile(r"^[\w-]+$")
```

File: tests/web/api/test_kb_dir.py [hunk 2]
```diff
@@ -254,6 +254,36 @@ def test_kb_ingest_rejects_invalid_characters_in_collection_name(
             assert "Invalid collection name" in response.json()["detail"]
 
 
+def test_kb_ingest_accepts_unicode_collection_name(test_env, temp_uploads):
```
Comment (rogercloud) [inline tests/web/api/test_kb_dir.py:260]: ### [Minor] Consider adding tests for Unicode names in other KB endpoints

This test covers the ingest flow well. It would be good to also add coverage for:

- Delete collection with Unicode name (`DELETE /collections/{collection_name}`)
- Rename collection with Unicode name (`PUT /collections/{collection_name}`)
- Check documents with Unicode collection name (`POST /collections/{collection_name}/documents/check`)
- Parse results with Unicode collection name (`GET /collections/{collection_name}/parses/...`)

This would ensure the full API surface handles Unicode consistently.

**Review by rogercloud — CHANGES_REQUESTED** [review]
## Review: Unicode Collection Names

The core regex change from `^[a-zA-Z0-9_-]+$` to `^[\w-]+$` is correct — `\w` in Python includes Unicode word characters while still rejecting spaces, dots, slashes, null bytes, etc. All LanceDB queries properly use `escape_lancedb_string()`, so no injection risk there.

However, I found several issues that need attention:

### Major (must-fix)

**1. Dead `validate_collection_name()` in RAG tools still rejects Unicode**

`src/xagent/core/tools/core/RAG_tools/utils/string_utils.py:232` still has:
```python
_COLLECTION_NAME_PATTERN: re.Pattern[str] = re.compile(r"^[a-zA-Z0-9_-]+$")
```

This is used by `validate_collection_name()` (line 236), which is called by `build_safe_collection_filter()` (line 311). While `build_safe_collection_filter` is currently dead code (never imported anywhere), a future developer wiring it up would **silently break** Unicode collection names. Either update the pattern to `[\w-]+` or remove the dead code entirely.

**2. Unsanitized `collection_name` path parameters in 3 API endpoints**

These endpoints receive `collection_name` as a FastAPI path parameter but never call `sanitize_path_component()`:

- `POST /collections/{collection_name}/documents/check` (`kb.py:1317`) — flows to LanceDB via `escape_lancedb_string()` (safe)
- `DELETE /collections/{collection_name}/documents/{filename}` (`kb.py:1395`) — flows to LanceDB (safe) **and** to `UploadedFile.storage_path.like(f"%{user_segment}%")` at line 1553 (unsanitized in LIKE pattern)
- `GET /collections/{collection_name}/parses/{doc_id}/parse_result` (`kb.py:1901`) — flows to LanceDB (safe)

While the LanceDB queries are injection-safe via escaping, the defense-in-depth principle requires validation at the API boundary. The `.like()` query in the delete endpoint is especially worth fixing. Follow the pattern used in `delete_collection_api` (line 1176) and `rename_collection_api` (line 1767).

### Minor (considerable)

**3. `MAX_COLLECTION_NAME_LENGTH = 100` counts code points, not bytes**

`config.py:93` — `len()` counts Unicode code points, but ext4 limits filenames to 255 bytes. 100 CJK characters = 300 UTF-8 bytes, exceeding the filesystem limit and causing `mkdir()` to fail with `OSError`. Pre-existing but now reachable. Consider switching to `len(name.encode('utf-8'))` for the byte count check.

**4. `get_file_url()` lacks URL encoding for Unicode collection names**

`config.py:269` interpolates raw Unicode into a URL string. Currently dead code (never called), but would be a bug if resurrected.

**5. Missing test coverage for Unicode names in other endpoints**

The new test only covers the ingest flow. No tests for delete, rename, check-documents, or parse-result endpoints with Unicode collection names.

**Comment by qinxuye** [comment]
White space should be allowed, but seems this PR did not address it, @yiboyasss

**Comment by yiboyasss** [comment]
Currently, collection names do not support spaces. I would recommend supporting arbitrary characters (including spaces) to improve usability and flexibility, as long as proper escaping/sanitization is ensured.

File: src/xagent/web/api/kb.py [hunk 3]
```diff
@@ -190,21 +190,28 @@ async def save_collection_config(
     """Save ingestion configuration for a specific collection."""
     from ...core.tools.core.RAG_tools.storage.factory import get_metadata_store
 
+    try:
```
Comment (rogercloud) [inline src/xagent/web/api/kb.py:193]: ### [Minor] Duplicate `sanitize_path_component` call

This `try/except` block duplicates the identical one at the top of the function (lines 1466–1473). The first call already assigns `safe_collection_name`, which is then overwritten here without being used in between.

Remove this second block (lines 1490–1497) — the validation already happened at function entry.

**Review by rogercloud — COMMENTED** [review]
## Follow-up: duplicate sanitization call

All original review comments have been addressed. One minor leftover found during re-review.

**Review by rogercloud — APPROVED** [review]
All review comments addressed and the duplicate sanitization is cleaned up. Tests pass (66/66). LGTM.
